File: C_Programs/factorial.c
#include <stdio.h>
#include <stdint.h>

unsigned long long factorial(unsigned int n) {
    unsigned long long r = 1;
    for (unsigned int i = 2; i <= n; ++i) r *= i;
    return r;
}

int main(void) {
    unsigned int n;
    printf("Enter a non-negative integer: ");
    if (scanf("%u", &n) != 1) {
        fprintf(stderr, "Invalid input\n");
        return 1;
    }
    printf("%u! = %llu\n", n, factorial(n));
    return 0;
}

File: Java_Programs/Factorial.java
import java.util.Scanner;

public class Factorial {
    public static long factorial(int n) {
        long r = 1;
        for (int i = 2; i <= n; i++) r *= i;
        return r;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a non-negative integer: ");
        if (!sc.hasNextInt()) { System.err.println("Invalid input"); sc.close(); System.exit(1); }
        int n = sc.nextInt();
        if (n < 0) { System.err.println("Negative number not allowed"); sc.close(); System.exit(1); }
        System.out.println(n + "! = " + factorial(n));
        sc.close();
    }
}
